src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -127,30 +150,100 @@ where
                 self.process_terminate();
             }
             FeMessage::Parse(m) => {
-                *unnamed_query_string = m.query_string;
+                // Step 1: Create the types description
+                let type_ids = m.type_ids;
+                let types: Vec<TypeOid> = type_ids
+                    .into_iter()
+                    .map(|x| TypeOid::as_type(x).unwrap())
+                    .collect();
+                // Step 2: Create the row description
+                let rows: Vec<PgFieldDescriptor> = types
+                    .iter()
+                    .map(|x| PgFieldDescriptor::new(String::new(), x.to_owned()))
+                    .collect();
+                // Step 3: Create the statement
+                let statement = PgStatement::new(
+                    cstr_to_str(&m.statement_name).unwrap().to_string(),
+                    m.query_string,
+                    types,
+                    rows,
+                );
+                // Step 4: Insert the statement
+                let name = statement.name();
+                if name.is_empty() {
+                    *unnamed_statement = statement;
+                } else {
+                    named_statements.insert(name, statement);
+                }
+                // println!("{}", cstr_to_str(&unnamed_query_string).unwrap());
                 self.write_message(&BeMessage::ParseComplete).await?;
             }
-            FeMessage::Bind(_) => {
+            FeMessage::Bind(m) => {
+                let statement_name = cstr_to_str(&m.statement_name).unwrap().to_string();
+                // Step 1 Get statement
+                let statement = if statement_name.is_empty() {
+                    unnamed_statement
+                } else {
+                    // NOTE Error handle method may need to modified
+                    named_statements.get(&statement_name).unwrap()
```

Comment:
Did you mean in this place we don't need this comment, we don't need extra handle, unwrap() is enough?

Reply:
I mean `named_statements.get(&statement_name)?`

Reply:
I realized a problem,  due to named_statements.get(&statement_name)->Option\<V\>, it can't directly use '?' to propagate the error . If try to change it to Err(), it should change it to a IO error. (Because the do_prooce return Result<bool,IO_error>),Doesn't that look weird? So should we just keep it 'unwrap()' or is there another better way to fix this problem?